roles/users/tasks/main.yml

```diff
@@ -50,7 +50,7 @@
   changed_when: false
   sudo: no
   register: admin_user_status
-  when: not sshd_permit_root_login
+  when: not sshd_permit_root_login and ansible_ssh_host is defined
```

Comment:
It does seem that _actually connecting_ over SSH to a host overrides `ansible_connection`. Ansible seems to prefer connecting over SSH if it can, even when `--connection=local` is set. So if Ansible knows how to SSH to `127.0.0.1`, it will, and will "undefine" `ansible_connection` accordingly. That seems like good behavior, even if Ansible perhaps should alert you that it's not using your preferred connection. Not sure if that's what is happening for you.